src/core/test/spec.js

```diff
@@ -104,64 +105,89 @@ describe('Test Passables "test" function', () => {
+
                         done();
                     }, 550);
                 });
 
                 it('Should pass for fulfilled promises', (done) => {
-                    expect(output.hasErrors(f3)).to.equal(false);
+                    expect(output.hasErrors(f4)).to.equal(false);
+
                     setTimeout(() => {
-                        expect(output.hasErrors(f3)).to.equal(false);
+                        expect(output.hasErrors(f4)).to.equal(false);
                         done();
                     }, 500);
                 });
             });
 
             describe('When passing a Promise as a test', () => {
                 describe('failing', () => {
-                    let f1, f2;
+                    let f1, f2, f3;
+                    const rejectionMessage = lorem.sentence();
+
                     beforeEach(() => {
                         f1 = lorem.word();
                         f2 = lorem.word();
+                        f3 = lorem.word();
 
-                        output = passable(lorem.word(), (test, draft) => {
-                            test(f1, lorem.sentence(), new Promise((resolve, reject) => setImmediate(reject)));
-                            test(f2, lorem.sentence(), new Promise((resolve) => setTimeout(resolve)));
-                            test(f2, lorem.sentence(), new Promise((resolve) => setTimeout(resolve, 500)));
+                        output = passable(lorem.word(), (test) => {
+                            test(f1, lorem.sentence(), new Promise((_, reject) => setImmediate(reject)));
+                            test(f2, new Promise((_, reject) => setImmediate(() => reject(rejectionMessage))));
+                            test(f3, lorem.sentence(), new Promise((resolve) => setTimeout(resolve)));
+                            test(f3, lorem.sentence(), new Promise((resolve) => setTimeout(resolve, 500)));
                             test(lorem.word(), lorem.sentence(), noop);
                         });
                     });
 
                     it('Should immediately register tests', () => {
-                        expect(output.testCount).to.equal(4);
+                        expect(output.testCount).to.equal(5);
                     });
 
                     it('Should run async test promise', (done) => {
                         passable(lorem.word(), (test) => {
-                            test(f1, lorem.sentence(), new Promise((resolve) => done()));
+                            test(f1, lorem.sentence(), new Promise(() => done()));
                             test(lorem.word(), lorem.sentence(), noop);
                         });
                     });
 
+                    it('Should fail with rejection message when provided', (done) => {
+                        setTimeout(() => {
+                            expect(output.getErrors(f2)).to.deep.equal([rejectionMessage]);
```

Comment:
The test will be more resilient if you add a number of errors for this field, and check that the field includes this error.